modules/custom/cu_share/cu_share.module

```diff
@@ -537,3 +537,69 @@ function cu_share_flush_caches() {
     menu_link_save($item);
   }
 }
+
+/**
+ * Implements hook_shortcode_info().
+ */
+function cu_share_shortcode_info() {
+  $shortcodes = array();
+  $shortcodes['share_on_twitter'] = array(
+    'title' => t('Share On Twitter'),
+    'description' => t('Create a link and quote for readers to share on Twitter.'),
+    'process callback' => 'cu_share_share_on_twitter',
+    'tips callback' => 'cu_share_share_on_twitter_tips',
+    'default settings' => array(),
+  );
+  return $shortcodes;
+}
+
+/**
+ * Tips function for share_on_twitter shortcode.
+ */
+function cu_share_share_on_twitter_tips($format, $long) {
+  $output = '[share_on_twitter]Text to share on Twitter[/share_on_twitter]';
+  return $output;
+}
+
+/**
+ * Process function for share_on_twitter shortcode.
+ */
+function cu_share_share_on_twitter($attrs, $content = NULL) {
+  extract(shortcode_attrs(array(
+  ), $attrs));
+  // Character count - 140, url, ellipsis
+  $character_count = 140;
+  $content = strip_tags($content);
+  if (arg(0) == 'node' && is_numeric(arg(1))) {
+    // Get the nid
+    $nid = arg(1);
+    $current_url = url('node/' . $nid, array('absolute' => TRUE, 'alias' => TRUE));
```

Comment:
I think the alias key should be removed. Sending URLs with the alias makes more sense to me than including the system path.